simulation/service.go

```diff
@@ -34,27 +33,37 @@ func NewSimulationService(config string) (onet.Simulation, error) {
 }
 
 // Setup creates the tree used for that simulation
-func (e *simulation) Setup(dir string, hosts []string) (
+func (s *SimulationService) Setup(dir string, hosts []string) (
 	*onet.SimulationConfig, error) {
 	sc := &onet.SimulationConfig{}
-	e.CreateRoster(sc, hosts, 2000)
-	err := e.CreateTree(sc)
+	s.CreateRoster(sc, hosts, 2000)
+	err := s.CreateTree(sc)
 	if err != nil {
 		return nil, err
 	}
 	return sc, nil
 }
 
+// Node can be used to initialize each node before it will be run
+// by the server. Here we call the 'Node'-method of the
+// SimulationBFTree structure which will load the roster- and the
+// tree-structure to speed up the first round.
+func (s *SimulationService) Node(config *onet.SimulationConfig) error {
+	index, _ := config.Roster.Search(config.Conode.ServerIdentity.ID)
```

Comment:
The second argument is not an error, but I can check for a negative index, which indicates an error.